packages/fixie/src/main.ts

```diff
@@ -89,6 +90,41 @@ const currentPath = path.dirname(fileURLToPath(import.meta.url));
 const packageJsonPath = path.resolve(currentPath, path.join('..', 'package.json'));
 const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'));
 
+function errorHandler(error: any) {
+  if (error instanceof FixieClientError) {
+    // Error from a REST API call.
+    const url = error.url;
+    if (error.statusCode == 401) {
+      term('❌ Could not authenticate to the Fixie API at ').green(`${url}\n`);
+      if (process.env.FIXIE_API_URL) {
+        term('Your ').green('FIXIE_API_URL')(' is set to ').green(process.env.FIXIE_API_URL)('\n');
+        term('Check to ensure that this is the correct API endpoint.\n');
+      }
+      if (process.env.FIXIE_API_KEY) {
+        term('Your ').green('FIXIE_API_KEY')(' is set to ').green(process.env.FIXIE_API_KEY.slice(0, 12))('...\n');
+        term('Check to ensure that this is the correct key.\n');
+      }
+    } else if (error.statusCode == 400) {
+      term('❌ Client made bad request to ').green(`${url}\n`);
+      term('Please check that you are running the latest version using ').green('npx fixie@latest -V')('\n');
+      term('The version of this CLI is: ').green(packageJson.version)('\n');
+    } else if (error.statusCode == 403) {
+      term('❌ Forbidden: ').green(`${url}\n`);
+    } else if (error.statusCode == 404) {
+      term('❌ Not found: ').green(`${url}\n`);
+    } else {
+      term('❌ Error accessing Fixie API at ').green(url)(': ')(error.message)('\n');
+    }
+    term.green(JSON.stringify(error.detail, null, 2));
+  } else {
+    term('❌ Error: ')(error.message)('\n');
+  }
+}
+
+function catchErrors(fn: (...args: any[]) => Promise<any>) {
+  return (...args: any[]) => fn(...args).catch(errorHandler);
```

Comment:
nit:

```
try {
   await fn(...args);
catch (ex) {
   errorHandler(ex);
}
```

would also catch errors if `fn` returns a promise but is not `async`.